grl/environment/tmaze_lib.py

```diff
@@ -0,0 +1,101 @@
+import numpy as np
+from .examples_lib import to_dict
+
+def tmaze(n: int, discount: float = 0.9):
+    """
+    Return T, R, gamma, p0 and phi for tmaze, for a given corridor length n
+    """
+    n_states = 2 * n  # corridors
+    n_states += 2  # start states
+    n_states += 2  # T junctions
+    n_states += 1  # terminal state
+
+    T_up = np.eye(n_states, n_states)
+    T_down = T_up.copy()
+    T_up[[-1, -2, -3], [-1, -2, -3]] = 0
+    T_down[[-1, -2, -3], [-1, -2, -3]] = 0
+
+    # If we go up or down at the junctions, we terminate
+    T_up[[-1 - 1, -2 - 1], [-1, -1]] = 1
```

Comment:
Should we keep the subtraction? I think you are showing it to show how we work backwards to the T junction states, but then wouldn't it be `[-1 - 1, -1 - 2]` ? I'm not quite sure how it is making it more clear tbh.